api/v2/echo.go

```diff
@@ -196,6 +199,63 @@ func (s *server) RWMutexFind(c echo.Context) error {
 	return nil
 }
 
+func (s *server) ConcurrencySearchIndex(c echo.Context) error {
+	valueStr := c.Param("index")
+	value, err := strconv.Atoi(valueStr)
+
+	ctx, cancel := context.WithCancel(c.Request().Context())
+	defer cancel()
+
+	if err != nil {
+		return echo.NewHTTPError(echo.ErrBadRequest.Code, "Invalid value")
+	}
+
+	var wg sync.WaitGroup
+	s.mutex.RLock()
+	index, ok := s.list.SearchConcurrently(ctx, cancel, &wg, value)
```

Comment:
Bot search by index and search by values run one function and the logic is the same.
Is there anything that i do not understand here?